CrimeMonitor/CrimeMonitor.cs

```diff
@@ -0,0 +1,849 @@
+            if (faction.EDDBID == null)
+            {
+                record.faction = Properties.CrimeMonitor.blank_faction;
+                record.system = null;
+                return;
+            }
+            record.Allegiance = faction.Allegiance ?? Superpower.None;
+            List<string> factionSystems = faction.presences.Select(p => p.systemName).ToList();
+            record.factionSystems = factionSystems;
+
+            // 'Home system' is not designated
+            if (homeSystem == null)
+            {
+                // Look first in saved home systems
+                if (homeSystems.TryGetValue(record.faction, out string result))
+                {
+                    homeSystem = result;
+                }
+                // Use sytem which is part of faction name. Otherwise, system with highest influence
+                else
+                {
+                    List<FactionPresence> presences = faction.presences.Where(p => factionSystems.Contains(p.systemName)).ToList();
+                    homeSystem = FindHomeSystem(record.faction, factionSystems)
+                        ?? presences.OrderByDescending(p => p.influence).First().systemName;
+                }
+            }
+            // If 'home system' is desiginated, check if system is part of faction presence
+            else if (factionSystems.Contains(homeSystem))
+            {
+                if (FindHomeSystem(record.faction, factionSystems) == null && !homeSystems.ContainsKey(record.faction))
+                {
+                    // Save home system if not part of faction name and not previously saved
+                    homeSystems.Add(record.faction, homeSystem);
+                }
+            }
+            // System not found, exit.
+            else { return; }
+
+            StarSystem factionStarSystem = StarSystemSqLiteRepository.Instance.GetOrFetchStarSystem(homeSystem);
+            record.system = homeSystem;
+
+            if (factionStarSystem != null)
+            {
+                // Filter stations within the faction system which meet the game version and landing pad size requirements
+                 string shipSize = EDDI.Instance?.CurrentShip?.size ?? "Large";
+                List<Station> factionStations = EDDI.Instance.inHorizons ? factionStarSystem.stations : factionStarSystem.orbitalstations
+                    .Where(s => s.stationservices.Count > 0 && s.LandingPadCheck(shipSize)).ToList();
+
+                // Prioritize controlled stations
+                List<Station> controlledStations = factionStations.Where(s => s.Faction.name == record.faction).ToList();
+                if (controlledStations.Count > 0 && controlledStations.Min(s => s.distancefromstar) < 10000)
+                {
+                    factionStations = controlledStations;
+                }
+
+                // Build list to find the faction station nearest to the main star
```

Comment:
Hmm... stations on planets require entering and exiting a gravity well (which can significantly slow you down). Maybe make an option to prefer orbital stations and only use surface stations if no orbital stations are available? Maybe allow users to specify a distance offset to apply to surface stations when looking for the nearest station?

Reply:
Dude, that's a lot of options you're suggesting ;-)

Reply:
If I have to choose between a surface station and an orbital station, I'm going to prefer the orbital station unless the surface station is **much** closer. But I'm not suggesting that we choose more than one (and maybe we choose none - I just wanted to float some ideas for consideration). ;-)

Reply:
Understood. How about we solicit comments from the community during the beta and go from there?

Reply:
I ended up doing this anyway.